slowfast/models/video_model_builder.py

```diff
@@ -68,9 +78,970 @@
+        
+        if cfg.SLOWFAST.AU_REDUCE_TF_DIM:
+            tf_stride = 2
+        else:
+            tf_stride = 1
+        tf_dim_reduction = 1
+
+        self.s1 = stem_helper.VideoModelStem(
+            dim_in=cfg.DATA.INPUT_CHANNEL_NUM,
+            dim_out=[
+                width_per_group, 
+                width_per_group // cfg.SLOWFAST.BETA_INV, 
+                width_per_group // cfg.SLOWFAST.AU_BETA_INV
+            ],
+            kernel=[
+                temp_kernel[0][0] + [7, 7], 
+                temp_kernel[0][1] + [7, 7], 
+                [temp_kernel[0][2] + [9, 1], temp_kernel[0][2] + [1, 9]],
+            ],
+            stride=[[1, 2, 2], [1, 2, 2], [[1, 1, 1], [1, 1, 1]]],
+            padding=[
+                [temp_kernel[0][0][0] // 2, 3, 3],
+                [temp_kernel[0][1][0] // 2, 3, 3],
+                [[temp_kernel[0][2][0] // 2, 4, 0], [temp_kernel[0][2][0] // 2, 0, 4]],
+            ],
+            stride_pool=[True, True, False],
+        )
+        
+        if self.FS_FUSION[0] or self.AFS_FUSION[0]:
+            self.s1_fuse = FuseAV(
+                # Slow
+                width_per_group,
+                # Fast
+                width_per_group // cfg.SLOWFAST.BETA_INV,
+                cfg.SLOWFAST.FUSION_CONV_CHANNEL_RATIO,
+                cfg.SLOWFAST.FUSION_KERNEL_SZ,
+                cfg.SLOWFAST.ALPHA,
+                # Audio
+                width_per_group // cfg.SLOWFAST.AU_BETA_INV,
+                cfg.SLOWFAST.AU_FUSION_CONV_CHANNEL_MODE,
+                cfg.SLOWFAST.AU_FUSION_CONV_CHANNEL_DIM,
+                cfg.SLOWFAST.AU_FUSION_CONV_CHANNEL_RATIO,
+                cfg.SLOWFAST.AU_FUSION_KERNEL_SZ,
+                cfg.SLOWFAST.AU_ALPHA // tf_dim_reduction,
+                cfg.SLOWFAST.AU_FUSION_CONV_NUM,
+                # Fusion connections
+                self.FS_FUSION[0],
+                self.AFS_FUSION[0],
+                # AVS
+                self.AVS_FLAG[0],
+                self.AVS_PROJ_DIM,
+                # nGPUs and shards
+                num_gpus=cfg.NUM_GPUS,
+                num_shards=cfg.NUM_SHARDS,
+            )
+        
+        slow_dim = width_per_group + \
+            (width_per_group // out_dim_ratio if self.FS_FUSION[0] else 0)
+        self.s2 = resnet_helper.ResStage(
+            dim_in=[
+                slow_dim,
+                width_per_group // cfg.SLOWFAST.BETA_INV,
+                width_per_group // cfg.SLOWFAST.AU_BETA_INV,
+            ],
+            dim_out=[
+                width_per_group * 4,
+                width_per_group * 4 // cfg.SLOWFAST.BETA_INV,
+                width_per_group * 4 // cfg.SLOWFAST.AU_BETA_INV,
+            ],
+            dim_inner=[
+                dim_inner, 
+                dim_inner // cfg.SLOWFAST.BETA_INV, 
+                dim_inner // cfg.SLOWFAST.AU_BETA_INV
+            ],
+            temp_kernel_sizes=temp_kernel[1],
+            stride=[1] * 3,
```

Comment:
Besides, here the `stride` means the stride of res1 of the audio pathway is 1. However in the paper *Downsampling in time-frequency space is performed by stride 2^2 convolution in the center (“bottleneck”) filter of the first residual block in each stage from res2 to res5*.  In the code, it seems that res2 is not included.